src/test/scala/temple/RegenerationFilterTest.scala

```diff
@@ -0,0 +1,68 @@
+package temple
+
+import java.io.{File, PrintWriter}
+
+import org.scalatest.{BeforeAndAfter, FlatSpec, Matchers}
+import temple.builder.project.Project
+import temple.detail.PoliceSergeantNicholasAngel
+import temple.generate.FileSystem
+
+import scala.reflect.io.Directory
+
+class RegenerationFilterTest extends FlatSpec with Matchers with BeforeAndAfter {
+
+  behavior of "RegenerationFilterTest"
+
+  private val testDirectory = "/tmp/test"
+
+  after {
+    val testDirectory = new File(this.testDirectory)
+    if (testDirectory.exists()) new Directory(testDirectory).deleteRecursively()
+  }
+
+  private def setupFileSystemWhole(): Unit = {
+    new File(testDirectory).mkdirs
+    val testFile = new File(testDirectory + "/test-file.go")
+    new PrintWriter(testFile).write("foo")
+    val setupGo = new File(testDirectory + "/setup.go")
+    new PrintWriter(setupGo).write("bar")
+  }
+
+  private def setupFileSystemEmpty(): Unit =
+    new File(testDirectory).mkdirs
+
+  it should "filter correctly when setup.go exists" in {
+    setupFileSystemWhole()
+    val output = RegenerationFilter.filter(
+      testDirectory,
+      RegenerationFilterTestData.simpleTestProjectWithConflict,
+      PoliceSergeantNicholasAngel,
+    )
+    output shouldBe Project(
+      Map(
+        FileSystem.File("", "test-file.go") -> "test-contents",
+      ),
+    )
+  }
+
+  it should "not filter when no files exist" in {
+    setupFileSystemEmpty()
+    val output = RegenerationFilter.filter(
+      testDirectory,
+      RegenerationFilterTestData.simpleTestProjectWithConflict,
+      PoliceSergeantNicholasAngel,
+    )
+    output shouldBe RegenerationFilterTestData.simpleTestProjectWithConflict
```

Comment:
`projectFilesShouldMatch()`, to get the diff displaying properly on error? And in a few other places too.